### PR TITLE
[Dreamfort] add supporting files and documentation

### PR DESCRIPTION
**Pull request by myk002**
- Adds dreamfort supporting files to `data/examples` and installs them to `hack/examples`
- Adds a new guide that documents all the files
- Updates help text in dreamfort.csv to point to the new file locations

depends on DFHack/scripts#317 for the `prioritize` doc label.

### CODE REVIEW

**Comment by myk002** [comment]
If you're ok with the file layout and documentation approach, we can merge this. I can update the individual files later based on forum discussion.

**Comment by lethosor** [comment]
Yep, I have no problem updating file contents at a later date - just taking a look at how the docs look now

File: docs/guides/examples-guide.rst [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+your preferences.
+
+The ``init/`` subfolder
+-----------------------
+
+The :source:`init/ <data/examples/init>` subfolder contains useful DFHack
+`init-files` that you can copy into your main Dwarf Fortress folder -- the same
+directory as ``dfhack.init``.
+
+:source:`onMapLoad_dreamfort.init <data/examples/init/onMapLoad_dreamfort.init>`
```
Comment (lethosor) [inline docs/guides/examples-guide.rst:19]: The fact that these links don't work until the PR is merged makes me want to make `:source:` a little smarter... but that's definitely out of scope here :)

**Review by lethosor — APPROVED** [review]
Thanks for putting this together!

**Pull request merged**